woocommerce/woocommerce-ios
# [Woo POS] Implement reader update in-progress view

Part of #13070 


## Description


This PR implements the SwiftUI view when the card reader software update is in-progress with the design matching the existing app (minus some style polishes which we'll work on later). There are 3 other failure views that I'm working on separately due to the size of changes.

## Testing information


The easiest way to simulate the reader update is by using the simulated Stripe card reader. In `StripeCardReaderService` L114 `Terminal.shared.simulatorConfiguration.availableReaderUpdate`, we can test this PR by setting the update type to required. For some reason, I couldn't find a way to simulate the optional update, as the only case I could find that triggers the update alert sets the `required` parameter to `true`:

https://github.com/woocommerce/woocommerce-ios/blob/47e1d4eebb7e5b8ff6c4aee41779538d0e3595c1/WooCommerce/Classes/ViewRelated/CardPresentPayments/CardReaderConnectionController.swift#L490-L493

I tested the optional case by manually setting the `required` parameter to `false` and compared the POS UI with the existing app.

Prerequisite: having a store eligible for POS

- Switch to a store in the prerequisite if needed
- Turn on the feature switch in Menu > Settings > Experimental Features > POS, if needed
- Go to the Menu tab --> the POS row should appear shortly
- Tap on the POS row
- Tap `Connect now` in the bottom bar --> the reader update in-progress UI should be shown as in the screenshots for the required update

## Screenshots


in-progress - required update | in-progress - optional update | completion (same for both)
-- | -- | --
<img width="588" alt="Screenshot 2024-06-17 at 8 25 32 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/9d254f38-b1cb-47fd-b15f-ef0cc50dd956"> | <img width="588" alt="Screenshot 2024-06-17 at 8 25 32 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/718a4245-4e1f-4364-9075-55ad125d1bff"> | <img width="588" alt="Screenshot 2024-06-17 at 8 25 45 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/80454462-4ca5-486e-a5ae-3dc61d483d8d">


https://github.com/woocommerce/woocommerce-ios/assets/1945542/cff1e3cb-162d-44ba-af6d-ef5398fc2f2d



---
